newhelm/single_turn_prompt_response.py

```diff
@@ -1,15 +1,28 @@
-from typing import Any, Dict, List, Type, TypeVar
+from typing import Any, Dict, List, Mapping, Optional, Type, TypeVar, Union
 
 from pydantic import BaseModel, Field
 from newhelm.annotation import Annotation
 
 from newhelm.placeholders import Prompt
 from newhelm.sut import SUTResponse
+from newhelm.typed_data import TypedData
 
 # TODO: This whole file assumes single turn. We'll either need to make it
 # more complicated, or make parallel structures for multi-turn.
 
-_InT = TypeVar("_InT")
+_BaseModelType = TypeVar("_BaseModelType", bound=BaseModel)
+_Context = Optional[Union[TypedData, str, Mapping]]
```

Comment:
Super minor and totally okay if you prefer `Union` and `Optional`, but since we're already requiring at least 3.10, you can also just use `|` now https://peps.python.org/pep-0604/

```
_Context = TypedData | str | Mapping | None
```

Reply:
I think it is plausible we can reduce that restriction with time, but this will be easy to change back if we need to. Done.